platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/GradleEnterprisePluginAutoAppliedStatus.java

```diff
@@ -16,6 +16,10 @@
 
 package org.gradle.internal.enterprise.impl;
 
+import org.gradle.internal.service.scopes.Scopes;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+@ServiceScope(Scopes.BuildTree.class)
 public class GradleEnterprisePluginAutoAppliedStatus {
```

Comment:
💭 This looks like some state as well, and not really like a service. I wonder if this works as expected when loading from the configuration cache. Any idea what we use it for and how we could check if everything works as expected here?

Reply:
We use it for determining whether the GE plugin was applied due to `--scan` and wouldn't have otherwise been applied. There should be existing tests for that which could be modified to test for CC compatibility.

Reply:
I suppose you do this check in the plugin code on application to decide whether to add some things to the build or not. That is fine, since it happens during configuration time and its result (aka the resulting configuration) will be cached. And the `--scan` command line argument is already a CC input, so it should work.

Can you point me to the tests? I'd expect them to be running already with CC as well.

Reply:
This one?

https://github.com/gradle/gradle/blob/ef6a429b1c630053333680434d823a769354f08d/platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginConfigIntegrationTest.groovy#L56-L71

Reply:
Yep, seems to fail on a hit: 
<img width="802" alt="image" src="https://github.com/gradle/gradle/assets/423186/9356a979-95f2-46d5-a8d9-29a9a83b7a78">

```
gradleEnterprisePlugin.serviceFactoryCreate.config.buildScanRequest = REQUESTED
gradleEnterprisePlugin.serviceFactoryCreate.config.autoApplied = false
gradleEnterprisePlugin.serviceFactoryCreate.config.taskExecutingBuild = true
gradleEnterprisePlugin.serviceFactoryCreate.buildState.buildStartedTime = 1694533444835
gradleEnterprisePlugin.serviceFactoryCreate.buildState.currentTime = 1694533445240
gradleEnterprisePlugin.serviceFactoryCreate.buildState.buildInvocationId = mrrmegpqkfhffps3dw6buljf7u
gradleEnterprisePlugin.serviceFactoryCreate.buildState.workspaceId = q52ut7no6bag5drocmnj53azcu
gradleEnterprisePlugin.serviceFactoryCreate.buildState.userId = h2depcj44vgjppvypl4zowousi
gradleEnterprisePlugin.serviceFactoryCreate.buildState.daemonScanInfo.numberOfBuilds = 3
gradleEnterprisePlugin.buildOperationNotificationListener.received = true
> Task :t UP-TO-DATE

BUILD SUCCESSFUL in 413ms
gradleEnterprisePlugin.endOfBuild.buildResult.failure = null
Configuration cache entry reused.
```